src/dmd/blockexit.d

```diff
@@ -487,8 +484,11 @@ int blockExit(Statement s, FuncDeclaration func, bool mustNotThrow)
                 result = BE.errthrow;
                 return;
             }
-            if (mustNotThrow)
+            const caught = isCaught(s.exp.type);
+            if (mustNotThrow && !caught)
```

Comment:
isCaught is always evaluated, but the result is only used when mustNotThrow is true.

Reply:
Cheap test goes first.

Reply:
Can you adjust this, @MoonlightSentinel? If this is the only thing left I can fix this so we have this merged to unblock https://github.com/dlang/dmd/pull/12383

Reply:
I'll rename the function because it has a required side effect (checking + flagging enclosing `TryCatchStatements`). The name is a relict from an earlier iteration.

> If this is the only thing left I can fix this so we have this merged to unblock #12383

Still need to figure out what change causes DMD to emit wrong code, i.e. if `blockExit` yields wrong results or if some other part relies on previously wrong results.

Reply:
Do you have any clues about this, @ibuclaw? Unblocking and getting https://github.com/dlang/dmd/pull/12383 in would be awesome marketing for D newbies.